django/db/models/fields/related.py

```diff
@@ -1051,6 +1108,35 @@ def _check_unique(self, **kwargs):
             else []
         )
 
+    def _has_related_models_with_db_cascading(self, model):
+        """
+        If the foreignkey parent has DB cascading and the Current model has non
+        db cascading return true
+        """
+        if not hasattr(model, "_meta"):
+            return False
+
+        on_delete = getattr(self.remote_field, "on_delete", None)
+        related_models = [
+            rel.related_model
+            for rel in model._meta.get_fields()
+            if rel.related_model and not rel.auto_created
+        ]
+        has_related_cascade_db = any(
+            any(
+                (
+                    isinstance(rel, ForeignKey)
+                    and hasattr(rel.remote_field, "on_delete")
+                    and rel.remote_field.on_delete in [DB_CASCADE, DB_SET_NULL]
+                    and on_delete in [CASCADE, SET_NULL]
```

Comment:
I think we can check `on_delete in [CASCADE, SET_NULL]` just after we create the `on_delete` variable (currently line 1119 above) instead of in this double-nested loop.